src/pages/index.js

```diff
@@ -1,65 +1,141 @@
+  photoSelector,
+  editAvatarPopupSelector,
+  profileAvatarSelector,
 }) => {
   const formValidators = {};
 
   const imagePopup = new PopupWithImage(popupImageContainer);
   imagePopup.setEventListeners();
 
+  const tripletenApi = new Api(APIdata);
+
+  const userInfo = new UserInfo({ nameSelector, aboutSelector, photoSelector });
+
   const handleImageClick = (cardData) => {
     imagePopup.open(cardData);
   };
 
+  const handleApiCallback = ({ action, endpoint }) => {
+    return tripletenApi.editCard(action, endpoint);
+  };
+
+  const popupConfirm = new PopupWithConfimration(confirmPopupSelector);
+  popupConfirm.setEventListeners();
+
+  const handleCardDelete = () => {
+    return popupConfirm.open();
+  };
+
   const createCard = (cardData) => {
-    const card = new Card(cardData, "#card", handleImageClick);
+    const card = new Card(
+      cardData,
+      "#card",
+      handleImageClick,
+      handleApiCallback,
+      handleCardDelete
+    );
     return card.generateCard();
   };
 
   const cardList = new Section(
     {
-      items: initialData,
+      items: [],
       renderer: (data) => {
         cardList.addItem(createCard(data));
       },
     },
     cardsContainer
   );
-  cardList.renderItems();
 
-  const userInfo = new UserInfo({ nameSelector, jobSelector });
+  const initialize = () => {
+    tripletenApi
+      .getUser()
+      .then((result) => {
+        userInfo.setUserInfo(result);
+      })
+      .catch((error) => console.error(error));
+
+    tripletenApi
+      .getCard()
+      .then((result) => {
+        cardList.setItems(result.reverse());
+        cardList.renderItems();
+      })
+      .catch((error) => {
+        console.error(error);
+      });
+  };
+
+  //for adding bulk of initial cards on loading page
+  //raw code for future feature
+  // const addCardsBulk = (cardsData) => {
+  //   cardsData.forEach((card) => {
+  //     tripletenApi
+  //       .addCard(card)
+  //       .then((response) => cardList.addItem(response))
+  //       .catch((error) => console.error(error));
+  //   });
+  // };
+
+  // addCardsBulk(initialData);
 
   const handlePopupEdit = (inputs) => {
-    userInfo.setUserInfo({
-      newName: inputs["edit-name"],
-      newJob: inputs["edit-job"],
-    });
+    const inputData = {
+      name: inputs["edit-name"],
+      about: inputs["edit-job"],
+    };
+    return tripletenApi
+      .editUser(inputData)
+      .then(() => userInfo.setUserInfo(inputData))
+      .catch((error) => console.error(error));
   };
 
   const handlePopupAdd = (inputs) => {
-    cardList.addItem(
-      createCard({ name: inputs["add-name"], link: inputs["add-link"] })
-    );
+    const inputData = {
+      name: inputs["add-name"],
+      link: inputs["add-link"],
+      avatar: userInfo.getUserInfo().avatar,
+    };
+    return tripletenApi
+      .addCard(inputData)
+      .then(() => cardList.addItem(createCard(inputData)))
+      .catch((error) => console.error(error));
```

Comment:
You don't need `.catch` if you have it inside `PopupWithForm` . The error should go till there